docs/rules/no-import-of-scoped-modules-during-ssr.md

```diff
@@ -0,0 +1,72 @@
+# Disallow static import of scoped modules during SSR (`lwc/no-import-of-scoped-modules-during-ssr`)
+
+During server-side rendering (SSR), these modules resolve to static values on the server and are updated to actual values on the client.
+
+## Rule details
+
+**Static imports of user-scoped modules:**
+
+Example of **incorrect** code:
+
+```js
+import { LightningElement } from 'lwc';
+import userId from '@salesforce/user/Id';
+import isGuest from '@salesforce/user/isGuest';
+
+export default class Foo extends LightningElement {
+    // This component should not import user-scoped modules during SSR
+}
+```
+
+Example of **correct** code:
```

Comment:
@khangsfdc What is the recommendation for replacing user-scoped modules if we warn against using dynamic imports?

Reply:
This is tough since data providers are the only alternative, but that is not directly consumable by end customers atm.

However, the warning for dynamic imports should just warn the customer that it does not effect SSR output. As long as they are aware of the CSR limitations, it may still be fine to suggest dynamic import here.

cc @kevinv11n

Reply:
@khangsfdc it is consumable by end customers through `{!User.x}` bundling expressions but the user DP does not preload its data currently.

Reply:
@nrobertdehault do they have to do anything to enable the binding expressions or is that available to every component within experience sites?

Reply:
How about this guidance:

> The recommended declarative solution is to use a [data provider](LINK TO PUBLIC DOCS) to fetch this information. The dynamic import approach below is an anti-pattern and should be avoided.